src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -59,6 +376,40 @@ def __init__(self, sc, *args):
             self.ml = sc._jvm.org.apache.sysml.api.MLContext(sc._jsc, monitorPerformance, setForcedSparkExecType)
         except Py4JError:
             traceback.print_exc()
+            
+    def start(self):
+        try:
+            prevGlobalMLContext = MLContext.globalMLContext
+        except AttributeError:
+            pass
+        self.numTabs = 0
+        self.sqlCtx = SQLContext(self.sc)
+        MLContext.globalMLContext = self
+        self.reset()
+        self.pydml = []
+        self.matrixOutput = []
+        
+    
+    def stop(self, vals):
+        if isinstance(vals, Matrix) or isinstance(vals, Scalar):
+            vals.registerAsOutput()
+        else:
+            for i in range(0, len(vals)):
+                if isinstance(vals[i], Matrix) or isinstance(vals[i], Scalar):
+                    vals[i].registerAsOutput()
```

Comment:
Is the `matrixOutput` attribute used anywhere else, except during when `stop` is called?